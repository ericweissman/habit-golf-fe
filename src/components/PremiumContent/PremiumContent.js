import React from 'react';

const PremiumContent = ({changeFilter}) => {
  return (
    <div className="premium-content">
      <div className="local-pros-area">
        <h4>lessons</h4>
        <p>connect with local pros in your area</p>
        <button>find a pro</button>
      </div>
      <div className="online-videos">
        <h4>premium content</h4>
        <p>access online tutorials</p>
        <button>subscribe</button>
      </div>
      <button className="back-to-home" onClick={() => changeFilter('home')}>back</button>
    </div>
  );
}

export default PremiumContent;
