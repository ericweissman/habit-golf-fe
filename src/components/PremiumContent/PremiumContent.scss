.premium-content {
  display: flex;
  flex-direction: column; 
  width: 275px;
  margin-top: 20px;
  border-radius: 10px;
  background-color: $masters-green;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  align-items: center;
  text-align: center;
  h4 {
    color: $masters-yellow;
    font-weight: 200;
    font-size: 25px;
  }
  p {
    color: white;
    font-weight: 200;
    font-size: 20px;
    padding: 10px;
  }

  button {
    width: 150px;
    height: 40px;
    background-color: #fff;
    font-size: 18px;
    font-weight: 200;
    border: none;
    border-radius: 50px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    color: #076652;
    background-color: #FEE715;
    outline: none;
    margin: 10px 0;
  }
  .back-to-home {
      width: 75px;
      margin-bottom: 25px;
      background-color: whitesmoke;
    }
}

.local-pros-area {
  padding: 15px 0 0 0;
}

.online-videos {
  padding: 15px 0;
}

