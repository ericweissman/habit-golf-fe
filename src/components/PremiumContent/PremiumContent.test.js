import React from 'react';
import PremiumContent from './PremiumContent';
import { shallow } from 'enzyme'

describe('PremiumContent', () => {
  let wrapper
  it('should correctly match the snapshot', () => {
    wrapper = shallow(<PremiumContent changeFilter={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })
})
