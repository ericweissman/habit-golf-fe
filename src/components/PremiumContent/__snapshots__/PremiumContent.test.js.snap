// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PremiumContent should correctly match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PremiumContent
    changeFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="local-pros-area"
        >
          <h4>
            lessons
          </h4>
          <p>
            connect with local pros in your area
          </p>
          <button>
            find a pro
          </button>
        </div>,
        <div
          className="online-videos"
        >
          <h4>
            premium content
          </h4>
          <p>
            access online tutorials
          </p>
          <button>
            subscribe
          </button>
        </div>,
        <button
          className="back-to-home"
          onClick={[Function]}
        >
          back
        </button>,
      ],
      "className": "premium-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              lessons
            </h4>,
            <p>
              connect with local pros in your area
            </p>,
            <button>
              find a pro
            </button>,
          ],
          "className": "local-pros-area",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "lessons",
            },
            "ref": null,
            "rendered": "lessons",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "connect with local pros in your area",
            },
            "ref": null,
            "rendered": "connect with local pros in your area",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "find a pro",
            },
            "ref": null,
            "rendered": "find a pro",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              premium content
            </h4>,
            <p>
              access online tutorials
            </p>,
            <button>
              subscribe
            </button>,
          ],
          "className": "online-videos",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "premium content",
            },
            "ref": null,
            "rendered": "premium content",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "access online tutorials",
            },
            "ref": null,
            "rendered": "access online tutorials",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "subscribe",
            },
            "ref": null,
            "rendered": "subscribe",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "back",
          "className": "back-to-home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "back",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="local-pros-area"
          >
            <h4>
              lessons
            </h4>
            <p>
              connect with local pros in your area
            </p>
            <button>
              find a pro
            </button>
          </div>,
          <div
            className="online-videos"
          >
            <h4>
              premium content
            </h4>
            <p>
              access online tutorials
            </p>
            <button>
              subscribe
            </button>
          </div>,
          <button
            className="back-to-home"
            onClick={[Function]}
          >
            back
          </button>,
        ],
        "className": "premium-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                lessons
              </h4>,
              <p>
                connect with local pros in your area
              </p>,
              <button>
                find a pro
              </button>,
            ],
            "className": "local-pros-area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "lessons",
              },
              "ref": null,
              "rendered": "lessons",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "connect with local pros in your area",
              },
              "ref": null,
              "rendered": "connect with local pros in your area",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "find a pro",
              },
              "ref": null,
              "rendered": "find a pro",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                premium content
              </h4>,
              <p>
                access online tutorials
              </p>,
              <button>
                subscribe
              </button>,
            ],
            "className": "online-videos",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "premium content",
              },
              "ref": null,
              "rendered": "premium content",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "access online tutorials",
              },
              "ref": null,
              "rendered": "access online tutorials",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "subscribe",
              },
              "ref": null,
              "rendered": "subscribe",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "back",
            "className": "back-to-home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "back",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
